### DISCUSSION TITLE
Support about trasmission of reports transaction user statistics and user

### DISCUSSION THREAD

**Discussion by mfilone** [phax/phase4#205] [Q&A] [answered]
Hi, we implemented both report for peppol reporting with the fantastic support of https://github.com/phax/peppol-reporting and implemented also our AP with phase4, before build and send reports with sbdh we also implemented smp client (from peppol-common) for the following receiverId (last updated) 0208:0848934496 but when we try to retrieve smp client for endpoint and certificate before sending we encounter problem with URI produced (with testbed all working fine), "http://B-566e2107cbf46df0295c9bb28be78495.iso6523-actorid-upis.acc.edelivery.tech.ec.europa.eu/iso6523-actorid-upis::9925:be0848934496/services/busdox-docid-qns::urn:fdc:peppol:transaction-statistics-report:1.0::TransactionStatisticsReport##urn:fdc:peppol.eu:edec:trns:transaction-statistics-reporting:1.0::1.0" we get a 404

we talking for production environment.
Can you help us to understand if somenthing is missing or we did something wrong ?

**Comment by phax** [comment]
Hi @mfilone - thanks for the positive feedback.
Question: where did you get the receiver `0208:0848934496` from?

The official [Peppol Reporting - Service Provider Operational Guideline](https://docs.peppol.eu/edelivery/guidelines/reporting/2023-12-01%20Peppol%20Reporting%20-%20SP%20Operational%20Guideline%20v1.0.pdf) from https://docs.peppol.eu/edelivery/ mentions `9925:be0848934496` as the one-and-only receiving participant ID for OpenPeppol.

**Reply by mfilone** [reply]
Thanks @phax, but we already tried with this receiver id 9925:be0848934496, and in a recently comunication we received the new future receiver id so we simply tried it 0208:0848934496. 
Now we retried with 9925:be0848934496, but without response and smp client produce the following URI: http://B-566e2107cbf46df0295c9bb28be78495.iso6523-actorid-upis.acc.edelivery.tech.ec.europa.eu/iso6523-actorid-upis%3A%3A9925%3Abe0848934496/services/busdox-docid-qns%3A%3Aurn%3Afdc%3Apeppol%3Atransaction-statistics-report%3A1.0%3A%3ATransactionStatisticsReport%23%23urn%3Afdc%3Apeppol.eu%3Aedec%3Atrns%3Atransaction-statistics-reporting%3A1.0%3A%3A1.0, any idea?

**Reply by phax** [reply]
The identifier will stay `9925:be0848934496` - the 0208 one will not be used.

And the SMP registrations are only available for production, but not for test, as the respective backend system is not yet available.

**Reply by mfilone** [reply]
@phax 
Clear, but we talk for production environment... in fact to build SMP client we current using :  

`  new SMPClient(Phase4PeppolSender.URL_PROVIDER, receiverID, ESML.DIGIT_PRODUCTION)
                .setSecureValidation(false)
                .setTrustStore(cryptoFactory.getTrustStore());`

the smp client work only for testbed? 

because if a check from peppol pratical partecipantId  for open peppol we found this smp istance 
http://smp.peppol.org/iso6523-actorid-upis::9925:be0848934496/services/busdox-docid-qns::urn:fdc:peppol:end-user-statistics-report:1.1::EndUserStatisticsReport##urn:fdc:peppol.eu:edec:trns:end-user-statistics-report:1.1::1.1.

**Reply by phax** [reply]
For the Testbed your SMP client needs to use `ESML.DIGIT_TEST`. For the real transmission your SMP client needs to use `ESML.DIGIT_PRODUCTION`. Also the receiver IDs are different.

I don't understand what error is remaining..... Please try to rephrase your question.

**Reply by mfilone** [reply]
@phax 
Ok i try to rephrase question: 
preface: 

- We currently sent TSR or USR for production environment from our production environment 
- Currently we implemented SMP client build for ESML.DIGIT_PRODUCTION  with default Phase4PeppolSender.URL_PROVIDER

observations: 
When the smp client tries to retrieve endpoint informations for receiver id 9925:be0848934496 (Production) it produce the following url 
`http://B-566e2107cbf46df0295c9bb28be78495.iso6523-actorid-upis.acc.edelivery.tech.ec.europa.eu/iso6523-actorid-upis%3A%3A9925%3Abe0848934496/services/busdox-docid-qns%3A%3Aurn%3Afdc%3Apeppol%3Atransaction-statistics-report%3A1.0%3A%3ATransactionStatisticsReport%23%23urn%3Afdc%3Apeppol.eu%3Aedec%3Atrns%3Atransaction-statistics-reporting%3A1.0%3A%3A1.0` but this url seems incorrect because we can't retrieve a response. 

So we would try to understand  if we need to call smp clients for production environment starting from receiver id or not? and if currently  Default URL_PROVIDER work or not.

**Reply by phax** [reply, marked as answer]
Thanks for repharsing. I think you have a bug in your statement. When using `ESML.DIGIT_PRODUCTION` you should never create a URL that starts with `ttp://B-566e2107cbf46df0295c9bb28be78495.iso6523-actorid-upis.acc.edelivery.tech.ec.europa.eu/` because the contains `.acc` is the indicator for the test system which can only be obtained with `ESML.DIGIT_TEST`.
So if you use http://B-566e2107cbf46df0295c9bb28be78495.iso6523-actorid-upis.edelivery.tech.ec.europa.eu/iso6523-actorid-upis%3A%3A9925%3Abe0848934496/services/busdox-docid-qns%3A%3Aurn%3Afdc%3Apeppol%3Atransaction-statistics-report%3A1.0%3A%3ATransactionStatisticsReport%23%23urn%3Afdc%3Apeppol.eu%3Aedec%3Atrns%3Atransaction-statistics-reporting%3A1.0%3A%3A1.0 instead (without the `.acc`) it looks fine too me :)

hth
